Lib/glyphsLib/builder/components.py

```diff
@@ -46,30 +46,36 @@ def to_ufo_components(self, ufo_glyph, layer):
         to_ufo_components_nonmaster_decompose(self, ufo_glyph, layer)
         return
 
-    # The same for color layers.
-    if (
-        not self.minimal
-        and layer.components
-        and layer.layerId != layer.associatedMasterId
-        and (layer._is_color_palette_layer())
-    ):
-        logger.warning(
-            f"Glyph '{ufo_glyph.name}': All components of the color layer "
-            f"'{layer.name}' will be decomposed."
-        )
-        to_ufo_components_nonmaster_decompose(self, ufo_glyph, layer)
-        return
-
     pen = ufo_glyph.getPointPen()
     for index, component in enumerate(layer.components):
+        component_name = component.name
+        if layer._is_color_palette_layer():
+            # Glyphs handles components for color layers in a special way. If
+            # the component glyph has color layers of its own, the component
+            # use the first color layer with the same color index, otherwise it
+            # fallback to the default layer. We try to do that here as well.
+            font = layer.parent.parent
+            component_glyph = font.glyphs[component_name]
+            color_layers = [
+                l for l in component_glyph.layers if l._is_color_palette_layer()
+            ]
+            for i, l in enumerate(color_layers):
+                if l._color_palette_index() == layer._color_palette_index():
+                    if l.layerId != l.associatedMasterId:
+                        # If it is not a master layer, we rename it in
+                        # _to_ufo_color_palette_layers(), so we reference the
+                        # same name here.
+                        component_name += f".color{i}"
+                    break
+            print(ufo_glyph.name, component_name)
```

Comment:
I presune this `print` was a leftover? I'll delete it

Reply:
ah you already did remove it, thank you!
https://github.com/googlefonts/glyphsLib/commit/896817d12798f9bddc374392913c0ecd1dd3dce7